amundsen_application/api/exceptions.py

```diff
@@ -0,0 +1,7 @@
+
+
+class MailClientNotImplemented(Exception):
```

Comment:
None for now in the public repo, but this creates dedicated place for them to all go when they need to be added. 

I'm not motivated to move the exceptions we've created for our custom use case here, because I don't want to set a precedent for putting any code that is not directly used by the public code into the public code -- because if everyone does that then we get bloated code. I also see the potential for folks to end up deleting it as "helpful cleanup" if they `Ctrl-F` something and see that it's not used in our public code.